diff --git a/core/shared/platform/include/platform_wasi_types.h b/core/shared/platform/include/platform_wasi_types.h
index ac1a95ea..8fb84419 100644
--- a/core/shared/platform/include/platform_wasi_types.h
+++ b/core/shared/platform/include/platform_wasi_types.h
@@ -33,7 +33,7 @@ extern "C" {
 
 /* There is no need to check the WASI layout if we're using uvwasi or libc-wasi
  * is not enabled at all. */
-#if WASM_ENABLE_UVWASI != 0 || WASM_ENABLE_LIBC_WASI == 0
+#if WASM_ENABLE_UVWASI != 0 || WASM_ENABLE_LIBC_WASI == 0 || _MSC_VER
 #define assert_wasi_layout(expr, message) /* nothing */
 #else
 #define assert_wasi_layout(expr, message) _Static_assert(expr, message)
